cmd/apply/apply.go

```diff
@@ -113,19 +113,28 @@ func (o *Options) run() error {
 		// If the transform does not exist, assume that the resource file is
 		// not needed and ignore for now.
 		_, tfStatErr := os.Stat(tfPath)
-		if errors.Is(tfStatErr, os.ErrNotExist) {
-			o.logger.Infof("resource file: %v is skipped due to no transform file: %v", f.Info.Name(), tfPath)
-			continue
+		if err != nil && !errors.Is(tfStatErr, os.ErrNotExist) {
+			// Some other error here err out
+			return err
 		}
 
-		transformfile, err := os.ReadFile(tfPath)
-		if err != nil {
+		// Set doc to the object, only update the file if the transfrom file exists
+		doc, err := f.Unstructured.MarshalJSON()
+		if err != nil && !errors.Is(tfStatErr, os.ErrNotExist) {
+			// Some other error here err out
```

Comment:
Shouldn't this always return err when err != nil? If err != nil, this means `MarshalJSON()` failed,  even if `tfPath` doesn't exist. In the case when err != nil and we're not returning err now (because the transform file doesn't exist), then later when we call JSONToYAML, that will fail, won't it? Or is that fine since we'll just return the err from that call? (although "can't transform nil input" may be less useful to return than whatever would have been returned here.

Reply:
I think that I am doing all the right things this diff is a little confusing though